pkg/oc/cli/admin/prune/imageprune/prune.go

```diff
@@ -1088,11 +1094,17 @@ func pruneISTagHistory(
 	prunableImageNodes map[string]*imagegraph.ImageNode,
 	keepYoungerThan time.Time,
 	streamName string,
-	imageStream *imageapi.ImageStream,
+	imageStream *imagev1.ImageStream,
 	tag string,
 ) (tagUpdated, tagDeleted bool) {
-	history := imageStream.Status.Tags[tag]
-	newHistory := imageapi.TagEventList{}
+	var history imagev1.NamedTagEventList
+	for _, t := range imageStream.Status.Tags {
```

Comment:
@soltysh this pattern repeats over and over in code... I would make this a helper, something like:

```
func IfHasTag(imageStream, tagName, func(t Tag) error) bool, error
```

Reply:
I'll open a followup for these helpers.